butterfly-slack/src/main/java/com/paypal/butterfly/slack/SlackNotifierConfig.java

```diff
@@ -0,0 +1,133 @@
+package com.paypal.butterfly.slack;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.paypal.butterfly.api.TransformationListener;
+import com.paypal.butterfly.api.TransformationRequest;
+import com.paypal.butterfly.api.TransformationResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import javax.annotation.PostConstruct;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.io.File;
+/**
+ * Observer that notifies the transformation requester
+ * via Slack when a transformation finishes.
+ *
+ * @author edhwang
+ */
+@Configuration
+public class SlackNotifierConfig {
+    private static final Logger logger = LoggerFactory.getLogger(SlackNotifierConfig.class);
+    @Bean
+    public TransformationListener slackNotification() {
+        return new TransformationListener() {
+
+            private String token;
+            private WebTarget target;
+            private String postSuccessJson;
+            private String postAbortJson;
+            private boolean active = false;
+
+            @PostConstruct
+            public void postConstruct() {
+                try {
+                    token = System.getProperty("slack.token");
+                    if (token == null) {
+                        logger.error("Unable to locate slack token to send notifications");
+                        return;
+                    }
+                    Path postSuccessPath = new File(this.getClass().getResource("/postSuccess.json").toURI()).toPath();
+                    Path postAbortPath = new File(this.getClass().getResource("/postAbort.json").toURI()).toPath();
+                    postSuccessJson = new String(Files.readAllBytes(postSuccessPath));
+                    postAbortJson = new String(Files.readAllBytes(postAbortPath));
+                    Client client = ClientBuilder.newClient();
+                    target = client.target("https://www.slack.com/api/");
+                    active = true;
+                } catch (IOException e) {
+                    logger.error("Unable to read the slack message template file from the path", e);
+                } catch (URISyntaxException e) {
+                    logger.error("Unable to read path of the slack message template file", e);
+                }
+            }
+
+            @Override
+            public void postTransformation(TransformationRequest transformationRequest, TransformationResult transformationResult) {
+                if (active && postSuccessJson != null) {
```

Comment:
Please remove `postSuccessJson != null` from here. Checking `active` is enough. If you believe there is value in checking `postSuccessJson != null`, then do it in the `postConstruct` method, right after you set `postSuccessJson`.